simply-rets-openhouses.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/*
+ * simply-rets-api-helper.php - Copyright (C) 2014-2015 SimplyRETS, Inc.
+ *
+ * This file provides a class that has functions for retrieving and parsing
+ * data from the remote retsd api.
+ *
+*/
+
+/* Code starts here */
+
+class SimplyRetsOpenHouses {
+
+    /**
+     * Generate markup /openhouses search response.
+     */
+    public static function openHousesSearchResults($search_response) {
+        $markup = "";
+
+        if(!array_key_exists("response", $search_response)) {
+            return $markup;
+        }
+
+        foreach($search_response["response"] as $idx=>$oh) {
+            $markup .= SimplyRetsOpenHouses::openHouseSearchResultMarkup($oh);
+        }
+
+        return $markup;
+    }
+
+    /**
+     * Generate markup for a single open house search result
+     */
+    public static function openHouseSearchResultMarkup($openhouse) {
+        $listing = $openhouse->listing;
+        $full_address = SrUtils::buildFullAddressString($listing);
+        $details_link = SrUtils::buildDetailsLink($listing);
+        $list_price_fmtd = '$' . number_format($listing->listPrice);
+        $listing_id = $listing->listingId;
+
+        $dummy = plugins_url( 'assets/img/defprop.jpg', __FILE__ );
+        $main_photo = !empty($listing->photos) ? $listing->photos[0] : $dummy;
+        $photo_style = "background-image:url('$main_photo');background-size:cover;";
+
+        $listing_office = $listing->office->name;
+        $listing_agent = $listing->agent->firstName . ' ' . $listing->agent->lastName;
+        $compliance_markup = SrUtils::mkListingSummaryCompliance(
+            $listing_office,
+            $listing_agent
+        );
+
+        date_default_timezone_set("America/Los_Angeles");
```

Comment:
We need to create an admin option to set this as we might not always be able to detect or handle various cases with 100% accuracy. Some MLS areas use UTC timestamps, others use local times, and some don't even provide full timestamps.

An admin option will be kind of an escape hatch for the user to make sure the times are shown correct. We can also have a "Don't modify openhouse timestamps" option.